src/logging.h

```diff
@@ -104,47 +105,34 @@ namespace BCLog {
     };
     constexpr auto DEFAULT_LOG_LEVEL{Level::Debug};
     constexpr size_t DEFAULT_MAX_LOG_BUFFER{1'000'000}; // buffer up to 1MB of log data prior to StartLogging
-    constexpr uint64_t RATELIMIT_MAX_BYTES{1024 * 1024}; // maximum number of bytes that can be logged within one window
+    constexpr uint64_t RATELIMIT_MAX_BYTES{1024 * 1024}; // maximum number of bytes per source location that can be logged within the RATELIMIT_WINDOW
+    constexpr auto RATELIMIT_WINDOW{1h}; // time window after which log ratelimit stats are reset
+    constexpr bool DEFAULT_LOGRATELIMIT{true};
 
-    //! Keeps track of an individual source location and how many available bytes are left for logging from it.
-    class LogLimitStats
+    //! Fixed window rate limiter for logging.
+    class LogRateLimiter : public std::enable_shared_from_this<LogRateLimiter>
```

Comment:
sorry, `std::enable_shared_from_this<LogRateLimiter>` was from an earlier approach, can be removed now
```suggestion
    class LogRateLimiter
```

Reply:
Fixed in 3d630c2544e19480268426cda245796d4ce34ac3, nice job spotting this.